server/data/data-jmap/src/main/java/org/apache/james/jmap/api/filtering/Rule.java

```diff
@@ -35,8 +35,8 @@
     public static class Id {
 
         public static Id of(String id) {
-            Preconditions.checkNotNull(id, "id should not be null");
```

Comment:
No more null handling?

Reply:
IMO: empty or null are handled in `StringUtils.isNotEmpty(id)` and its should the same `invalidArgument`